### ISSUE TITLE
Not installing on Apple Silicon

### ISSUE THREAD

**Issue by simon-tiger** [BepInEx/BepInEx#513]
### What happened?

I tried to follow the installation instructions [here](https://docs.bepinex.dev/articles/user_guide/installation/index.html), then I ran `./run_bepinex.sh`. I got this error:
```
/Users/simontiger/Library/Application Support/Steam/steamapps/common/Patrick's Parabox/Patrick's Parabox.app/Contents/MacOS/Patrick's Parabox
dyld[3257]: terminating because inserted dylib '/Users/simontiger/Library/Application Support/Steam/steamapps/common/Patrick's Parabox/doorstop_libs/libdoorstop_x64.dylib' could not be loaded: tried: '/Users/simontiger/Library/Application Support/Steam/steamapps/common/Patrick's Parabox/doorstop_libs/libdoorstop_x64.dylib' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64e')), '/Users/simontiger/Library/Application Support/Steam/steamapps/common/Patrick's Parabox/doorstop_libs/libdoorstop_x64.dylib' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64e')), '/usr/local/lib/libdoorstop_x64.dylib' (no such file), '/usr/lib/libdoorstop_x64.dylib' (no such file)
dyld[3257]: tried: '/Users/simontiger/Library/Application Support/Steam/steamapps/common/Patrick's Parabox/doorstop_libs/libdoorstop_x64.dylib' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64e')), '/Users/simontiger/Library/Application Support/Steam/steamapps/common/Patrick's Parabox/doorstop_libs/libdoorstop_x64.dylib' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64e')), '/usr/local/lib/libdoorstop_x64.dylib' (no such file), '/usr/lib/libdoorstop_x64.dylib' (no such file)
./run_bepinex.sh: line 145:  3257 Abort trap: 6           "${executable_path}"
```

I interpret this part here:
```
(mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64e'))
```
as this having to do with my computer being on Apple silicon (my computer is an M1 mac mini), but if you think I'm wrong, let me know.

### BepInEx Distribution

Stable from GitHub

### Log outputs

```txt
/Users/simontiger/Library/Application Support/Steam/steamapps/common/Patrick's Parabox/Patrick's Parabox.app/Contents/MacOS/Patrick's Parabox
dyld[3257]: terminating because inserted dylib '/Users/simontiger/Library/Application Support/Steam/steamapps/common/Patrick's Parabox/doorstop_libs/libdoorstop_x64.dylib' could not be loaded: tried: '/Users/simontiger/Library/Application Support/Steam/steamapps/common/Patrick's Parabox/doorstop_libs/libdoorstop_x64.dylib' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64e')), '/Users/simontiger/Library/Application Support/Steam/steamapps/common/Patrick's Parabox/doorstop_libs/libdoorstop_x64.dylib' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64e')), '/usr/local/lib/libdoorstop_x64.dylib' (no such file), '/usr/lib/libdoorstop_x64.dylib' (no such file)
dyld[3257]: tried: '/Users/simontiger/Library/Application Support/Steam/steamapps/common/Patrick's Parabox/doorstop_libs/libdoorstop_x64.dylib' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64e')), '/Users/simontiger/Library/Application Support/Steam/steamapps/common/Patrick's Parabox/doorstop_libs/libdoorstop_x64.dylib' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64e')), '/usr/local/lib/libdoorstop_x64.dylib' (no such file), '/usr/lib/libdoorstop_x64.dylib' (no such file)
./run_bepinex.sh: line 145:  3257 Abort trap: 6           "${executable_path}"
```

### Environment

```markdown
- OS: MacOS 12.4 Monterey 
- BepInEx: 5.4.21 (latest stable release)
- Game: Patrick's Parabox (from Steam not itch.io)
```

**Comment by DiederikvandenB** [comment]
Same... Any update on this?

**Comment by simon-tiger** [comment]
No updates, still just as stuck as I was before

**Comment by smndtrl** [comment]
Same problem for me but I assume there is and won't be a immediate solution as any ARM chips are unsupported from looking at the support matrix on this repos README.

**Comment by IzaacJ** [comment]
I've had this issue for the past days on my M2 MacBook Air, so I've spent the last 2-3 hours getting a partial temporary workaround for it, by wrapping a lot of stuff in a forced-Rosetta command.

BepinEx loads, creates its directories and loads mod manager, but mod manager seems to crash occasionally.
I have not had time to check how it is working with other mods, so use at your mileage may vary.
I will have more time to test and hopefully improve it tomorrow evening.

My `run_bepinex.sh` has had the last line (`"${executable_path}" "$@"`) replaced with this humongous piece of code that wraps a lot of the stuff for Rosetta.

```
current_path=$(pwd)
exports="export LD_LIBRARY_PATH=\"$LD_LIBRARY_PATH\";export LD_PRELOAD=$LD_PRELOAD;export DYLD_LIBRARY_PATH=\"$DYLD_LIBRARY_PATH\";export DYLD_INSERT_LIBRARIES=\"$DYLD_INSERT_LIBRARIES\""
cdir="cd \"$current_path\""
exec="\"${executable_path}\""
a="\"$@\""
launch="arch;$cdir;pwd;$exports;$exec $a"
echo "Launch Command: $launch"
arch -x86_64 zsh -c "$launch"
```

Usage:

1. Open Terminal.app
2. Change to `/Users/<username>/Library/Application Support/Steam/steamapps/common/Timberborn`
3. Run ´./run_bepinex.sh`
4. Hopefully success!

**Comment by allquixotic** [comment]
> I've had this issue for the past days on my M2 MacBook Air on Sonoma 14.1.1, so I've spent the last 2-3 hours getting a partial temporary workaround for it, by wrapping a lot of stuff in a forced-Rosetta command.
> 
> BepinEx loads, creates its directories and loads mod manager. I have not had time to check how it is working with other mods, so your mileage may vary. I will have more time to test and hopefully improve it tomorrow evening.
> 
> My `run_bepinex.sh` has had the last line (`"${executable_path}" "$@"`) replaced with this humongous piece of code that wraps a lot of the stuff for Rosetta.
> 
> ```
> current_path=$(pwd)
> exports="export LD_LIBRARY_PATH=\"$LD_LIBRARY_PATH\";export LD_PRELOAD=$LD_PRELOAD;export DYLD_LIBRARY_PATH=\"$DYLD_LIBRARY_PATH\";export DYLD_INSERT_LIBRARIES=\"$DYLD_INSERT_LIBRARIES\""
> cdir="cd \"$current_path\""
> exec="\"${executable_path}\""
> a="\"$@\""
> launch="arch;$cdir;pwd;$exports;$exec $a"
> echo "Launch Command: $launch"
> arch -x86_64 zsh -c "$launch"
> ```
> 
> Usage:
> 
>     1. Open Terminal.app
> 
>     2. Change to `/Users/<username>/Library/Application Support/Steam/steamapps/common/Timberborn`
> 
>     3. Run ´./run_bepinex.sh`
> 
>     4. Hopefully success!
> 
> 
> EDIT: This came about after some troubleshooting on [discord](https://discord.com/channels/558398674389172225/1064824959697944666/1172680496052982013) with a person that had no issues what-so-ever on his M1 Mac on Sonoma 14.1.1.
> 
> EDIT2: Quickly did some surface-level testing with the following mods, and they all seem to work as expected:
> 
>     * Mod Manager
> 
>     * TimberAPI
> 
>     * Staircase
> 
>     * Hide Welcome Window
> 
>     * Simple Floodgate Triggers

This is very useful for folks trying to run Valheim on MacOS as well. Thank you for the tip on how to run BepInEx in forced Intel architecture mode.

**Comment by ckanitz** [comment]
Well, this helped me to at least start the game via the bepinex start script, but when I add dll files to the bepinex/plugins folder (== adding mods) the game will just crash after the splash screens
if I want to start it via steam I need to remove `%command%` after `./start_game_bepinex.sh` so the game start wont just error instantly
however: even when I somehow manage to start the game at least into the start screen, no info that bepinix is loaded / "a modded version of the games" is run.

In conclusion: The game is somewhat startable via the bepinex start script but wont load any mods atm :|


Tested on: M1 | MacOS 14

**Comment by allquixotic** [comment]
> Well, this helped me to at least start the game via the bepinex start script, but when I add dll files to the bepinex/plugins folder (== adding mods) the game will just crash after the splash screens if I want to start it via steam I need to remove `%command%` after `./start_game_bepinex.sh` so the game start wont just error instantly however: even when I somehow manage to start the game at least into the start screen, no info that bepinix is loaded / "a modded version of the games" is run.
> 
> In conclusion: The game is somewhat startable via the bepinex start script but wont load any mods atm :|
> 
> Tested on: M1 | MacOS 14

Which game?

If you're trying to play Valheim, here is a modified version of the start script that works with the current stable release of BepInEx: https://gist.github.com/allquixotic/0530bde2247415a676288e8f62592a4d

**Comment by ckanitz** [comment]
thanks you, I adapted the script but the game still crashes with any mod present in the `/Valheim/BepinEx/plugins`
wont crash if no plugins are present but won't show that it's a modded game version (as I'm used to via linux)

**Comment by ckanitz** [comment]
Alright I got it working:
- Deleted everything but `valheim.app` from `/steamapps/common/Valheim`
- uninstalled Valheim via Steam
- reinstalled Valheim via Steam
- created `BepInEx` folder manually within `/Valheim`
- created `plugins` folder manually within `BepInEx`
- copied my set of mods into the `plugins` folder
- followed this guide: https://www.reddit.com/r/valheim/comments/1dcko3i/guide_running_mods_on_macos/
- - downloaded the mentioned bepinex version
- - copy + merged(!) to `/Valheim`
- - replaced contents of `run_bepinex.sh` with the linked gist
- - ran `chmod +x run_bepinex.sh`
- - ran `./run_bepinex.sh`

voilá

**Comment by allquixotic** [comment]
Cool, now you're all caught up with me. :) 

I created this https://github.com/BepInEx/BepInEx/issues/899 but it looks like the real problem isn't with BepInEx, but with UnityDoorstop. Therefore the real issue to track is https://github.com/NeighTools/UnityDoorstop/issues/61
